webapi/src/main/scala/org/knora/webapi/messages/StringFormatter.scala

```diff
@@ -2007,49 +2034,63 @@ class StringFormatter private (
       case None    => None
     }
 
+  def validateAndEscapeOptionalProjectShortname(
+    maybeString: Option[String]
+  ): Validation[ValidationException, Option[String]] =
+    maybeString match {
+      case None        => Validation.succeed(None)
+      case Some(value) => validateAndEscapeProjectShortname(value).map(Some(_))
+    }
+
   /**
    * Given the project shortcode, checks if it is in a valid format, and converts it to upper case.
    *
    * @param shortcode the project's shortcode.
    * @return the shortcode in upper case.
    */
+  @deprecated("Use def validateProjectShortcode(String) instead.")
   def validateProjectShortcode(shortcode: String, errorFun: => Nothing): String = // V2 / value objects
-    ProjectIDRegex.findFirstIn(shortcode.toUpperCase) match {
-      case Some(value) => value
-      case None        => errorFun
-    }
+    validateProjectShortcode(shortcode).getOrElse(errorFun)
+
+  def validateProjectShortcode(shortcode: String): Option[String] =
+    ProjectIDRegex.findFirstIn(shortcode.toUpperCase)
 
+  @deprecated("Use escapeOptionalString(Option[String]) instead.")
   def escapeOptionalString(maybeString: Option[String], errorFun: => Nothing): Option[String] = // --
     // TODO: I leave this for now to avoid merge conflicts. Should be moved to the ValuesValidator as soon as possible. (depends on toSparqlEncodedString())
-    maybeString match {
-      case Some(s) =>
-        Some(toSparqlEncodedString(s, errorFun))
-      case None => None
-    }
+    maybeString.map(toSparqlEncodedString).getOrElse(errorFun)
+
+  def escapeOptionalString(maybeString: Option[String]): Option[String] =
+    maybeString.flatMap(toSparqlEncodedString)
```

Comment:
this doesn't look right: here, in the response None is an illegal state (so should be Validation IMO), but in the input it's legal, so when passing in None, it would wrongly fail, I think.  
I think this should return `Validation[ValidationException. Option[String]]` to make it clear what's legal and what not

Reply:
All "optional" methods are gone now 🔥